### DISCUSSION TITLE
ImageMagick newbie needs help doing multiple "convert -distort SRT"s in one invocation of Magick

### DISCUSSION THREAD

**Discussion by ItzRav** [ImageMagick/ImageMagick#3973] [Help] [answered]
Using ImageMagick under Windows 10.  I just started using it today and need some assistance.  I have a single input file which contains a circle image.  I need to create 360 separate output files based on that input file, where each successive file contains the original image rotated 1 degree, then 2 degrees, then 3 ... up to 360 (well, 359 actually, since the original image is at 0 degrees).  I need to use "-distort SRT" instead of rotate since I had some issues with rotate.  "-distort SRT" works perfectly for me.  At any rate, I currently have a batch file that looks like this:

@echo off
Magick convert Wheel0.bmp -distort SRT -1 Wheel1.bmp
Magick convert Wheel0.bmp -distort SRT -2 Wheel2.bmp
Magick convert Wheel0.bmp -distort SRT -3 Wheel3.bmp
.
.
.
Magick convert Wheel0.bmp -distort SRT -357 Wheel357.bmp
Magick convert Wheel0.bmp -distort SRT -358 Wheel358.bmp
Magick convert Wheel0.bmp -distort SRT -359 Wheel359.bmp

I realize this is very inefficient.  I have looked at multiple web sites trying to understand how to create a single call to Magick convert which would specify the one input file (possibly caching it in memory for additional efficiency) and the 359 output files, and I can't understand how to do it.  I am NOT looking for a FOR-type statement which would have the batch file loop -- that would still call Magick convert hundreds of times.  I'd appreciate any help.  Thanks.

**Comment by fmw42** [comment]
Windows Syntax

```
magick Wheel0.bmp -write mpr: +delete ^
( mpr: -distort SRT -1 +write Wheel1.bmp ) ^
( mpr: -distort SRT -2 +write Wheel2.bmp ) ^
...
( mpr: -distort SRT -359 +write Wheel359.bmp ) ^
null:

```

See https://legacy.imagemagick.org/Usage/files/#write

Use magick, not magick convert. Possibly need magick.exe on Windows?

**Reply by ItzRav** [reply]
Thank you!  It works great.

**Comment by snibgo** [comment, marked as answer]
fmw42's method should work fine, but will eat memory. After each output is written, we don't need it any more, so we can "+delete" it. Instead of "+write", "-write" may be faster, and is just as good in this example. And I would always name the "mpr:", eg "mpr:INP".
```
magick Wheel0.bmp -write mpr:INP +delete ^
( mpr:INP -distort SRT -1 -write Wheel1.bmp +delete ) ^
( mpr:INP -distort SRT -2 -write Wheel2.bmp +delete ) ^
...
( mpr:INP -distort SRT -359 -write Wheel359.bmp +delete ) ^
null:
```

**Reply by fmw42** [reply]
Thanks snibgo. I forgot the delete. I should have written that.

**Reply by ItzRav** [reply]
Thank you too.  I'll give that a try tomorrow.

**Reply by ItzRav** [reply]
Thanks!  That DEFINITELY uses a LOT less memory!  Thanks to both you and fmw42 (I wish I could mark both your replies as "the answer").
